src/App.tsx

```diff
@@ -1,14 +1,73 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import debounce from 'lodash.debounce';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [loaded, setLoaded] = useState(true);
+  const [filterStatus, setFilterStatus] = useState('all');
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  useEffect(() => {
+    const loadTodos = async () => {
+      const loadedTodos = getTodos();
```

Comment:
The function `getTodos()` should be awaited directly. Consider using `await getTodos();` instead of assigning it to a variable first.